crates/storage/db-common/src/lib.rs

```diff
@@ -8,4 +8,192 @@
 #![cfg_attr(not(test), warn(unused_crate_dependencies))]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
 
+use boyer_moore_magiclen::BMByte;
+use eyre::Result;
+use reth_chainspec::ChainSpec;
+use reth_db::{RawTable, TableRawRow};
+use reth_db_api::{
+    cursor::{DbCursorRO, DbDupCursorRO},
+    database::Database,
+    table::{Decode, Decompress, DupSort, Table, TableRow},
+    transaction::{DbTx, DbTxMut},
+    DatabaseError,
+};
+use reth_fs_util as fs;
+use reth_provider::{ChainSpecProvider, ProviderFactory};
+use std::{path::Path, rc::Rc, sync::Arc};
+use tracing::info;
+
 pub mod init;
+
+/// Wrapper over DB that implements many useful DB queries.
+#[derive(Debug)]
+pub struct DbTool<DB: Database> {
+    /// The provider factory that the db tool will use.
+    pub provider_factory: ProviderFactory<DB>,
```

Comment:
this should be moved to its own module

Reply:
Just fixed, let me know what you think about this. Just put `DbTool` inside a mod file. Imo there is too little to implement here to create a separate file next to the mod.rs file